pkg/apis/acme/v1alpha3/types_issuer.go

```diff
@@ -23,30 +23,51 @@ import (
 	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
 )
 
-// ACMEIssuer contains the specification for an ACME issuer
+// ACMEIssuer contains the specification for an ACME issuer.
+// This uses the RFC8555 specification to obtain certificates by completing
+// 'challenges' to prove ownership of domain identifiers.
+// Earlier draft versions of the ACME specification are not supported.
 type ACMEIssuer struct {
-	// Email is the email for this account
+	// Email is the email address to be associated with the ACME account.
+	// This field is optional, but it is strongly recommended to be set.
+	// It will be used to contact you in case of issues with your account or
+	// certificates, including expiry notification emails.
+	// This field may be updated after the account is initially registered.
 	// +optional
 	Email string `json:"email,omitempty"`
 
-	// Server is the ACME server URL
+	// Server is the URL used to access the ACME server's 'directory' endpoint.
+	// For examples, for Let's Encrypt's staging endpoint, you would use:
+	// "https://acme-staging-v02.api.letsencrypt.org/directory".
+	// Only ACME v2 endpoints (i.e. RFC 8555) are supported.
 	Server string `json:"server"`
 
-	// If true, skip verifying the ACME server TLS certificate
+	// If true, requests to the ACME server will not have their TLS certificate
+	// validated (i.e. insecure connections will be allowed).
+	// Only enable this option in development environments.
+	// The cert-manager system installed roots will be used to verify connections
+	// to the ACME server if this is false.
+	// Defaults to false.
 	// +optional
 	SkipTLSVerify bool `json:"skipTLSVerify,omitempty"`
 
 	// ExternalAccountBinding is a reference to a CA external account of the ACME
 	// server.
+	// If set, upon registration cert-manager will attempt to associate the given
+	// external account credentials with the registered ACME account.
 	// +optional
 	ExternalAccountBinding *ACMEExternalAccountBinding `json:"externalAccountBinding,omitempty"`
 
-	// PrivateKey is the name of a secret containing the private key for this
-	// user account.
+	// PrivateKey is the name of a Secret resource that will be used to store the
+	// automatically generated ACME account private key.
+	// If `key` is not specified, a default of `tls.key` will be used.
 	PrivateKey cmmeta.SecretKeySelector `json:"privateKeySecretRef"`
 
 	// Solvers is a list of challenge solvers that will be used to solve
 	// ACME challenges for the matching domains.
+	// Solver configurations must be provided in order to obtain certificates
+	// from an ACME server.
+	// For more information, see: https://cert-manager.io/docs/configuration/acme/
```

Comment:
Maybe this would be a better place to explain that solvers are used in order of DNS specificity.
Currently this is explained below on the `ACMEChallengeSolver > Selector` field